pkg/storage/ingest/writer_test.go

```diff
@@ -395,9 +393,13 @@ func createTestWriter(t *testing.T, cfg KafkaConfig) (*Writer, prometheus.Gather
 }
 
 func createTestKafkaClient(t *testing.T, cfg KafkaConfig) *kgo.Client {
-	metrics := kprom.NewMetrics("", kprom.Registerer(prometheus.NewPedanticRegistry()))
+	opts := commonKafkaClientOptions(cfg, nil, test.NewTestingLogger(t))
```

Comment:
why remove the metrics as well?

Reply:
In past we didn't support `nil` as metrics. In one of my recent PRs, I've made `metrics` param optional (you can pass `nil` instead). In this PR I removed it because we just pass a `NewPedanticRegistry()` here, so we never look at these metrics, but I'm going to re-introduce: it could spot issues like duplicated metrics registered.

Reply:
Done in https://github.com/grafana/mimir/pull/7741/commits/d070f61ce90a96a4f49a9a78714836638b7c7583